File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        stack.add(arr[0]);
        
        for(int i=1; i<arr.length; i++) {
        	if(stack.peek() != arr[i]) stack.add(arr[i]);
        }
        int[] answer = new int[stack.size()];
        
        int i = answer.length - 1;
        while(!stack.isEmpty()) {
        	answer[i--] = stack.pop();
        }

        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    stack = []
    
    arr.reverse()
    stack.append(arr[0])
    
    for a in arr[1:]:
      if stack[-1] != a:
        stack.append(a)
        
    while(len(stack) > 0):
      answer.append(stack.pop())
    return answer